### DISCUSSION TITLE
How to get all classed discovered by Symfony during container compile/build phrase?

### DISCUSSION THREAD

**Discussion by Warxcell** [symfony/symfony#46666] [Q&A]
Use-case is that I need to discover all Enums tagged with specific attribute. `registerAttributeForAutoconfiguration` doesn't work, because enums are not real services/ they are not instantiable. 

Currently I'm doing the following hack in compiler pass:

```
$reflection = new ReflectionObject($container);
$classReflectors = $reflection->getProperty('classReflectors');
$classReflectors->setAccessible(true);
$reflectors = $classReflectors->getValue($container);

foreach ($reflectors as $reflector) {
    .. search for class with that attribute
}
```

**Comment by stof** [comment]
Well, implement a discovery system finding classes.
The container builder is focused on service definitions, not on finding available classes/enums.

**Reply by Warxcell** [reply]
But it already have all classes in `classReflectors` property, why should I re-invent the wheel?

**Reply by stof** [reply]
There is strictly no guarantee that it has all of them in that property (that property is only an internal cache)

**Reply by Warxcell** [reply]
Well it should have them all, because symfony doesn't know if class is enum/interface/abstract/etc until it loads it and create reflection for it.

**Reply by stof** [reply]
well, this cache has all the classes for which the container builder has previously created a reflector. There is no guarantee that this is all the classes available in your project (and there won't ever be such guarantee).
For instance, when your PSR-4 discovery excludes some files from the service discovery, it is very unlikely that classes defined in those files will appear in that cache.

**Reply by Warxcell** [reply]
That's fine - only PSR-0/PSR-4 discoverable and defined in symfony configs like that

```
App\:
    resource: '../src/*'
```

classes are wanted

**Reply by Warxcell** [reply]
Furthermore I may need all services that have specific attribute or implements interface that have that same attribute.
